### ISSUE TITLE
Disconnect throws an indexedDB error

### ISSUE THREAD

**Issue by AlexandrosGounis** [airgap-it/beacon-sdk#756]
**Describe the bug (current behavior)**
The latest demo DApp ([link](https://8452a04057e308e40c9f01c821490fca36bef9f0--beacon-sdk.netlify.app/dapp)) throws the following error when the `Disconnect` button is clicked:

```
IndexedDBStorage.js:73 Uncaught DOMException: Failed to execute 'transaction' on 'IDBDatabase': One of the specified object stores was not found.
```

**To Reproduce**
1. Go to the latest [example DApp deployment](https://8452a04057e308e40c9f01c821490fca36bef9f0--beacon-sdk.netlify.app/dapp)
2. Click `Request Permission`, accept the modal (all wallet-connect and p2p wallet options would do)
3. Click `Disconnect`

An indexedDB error is thrown in the console, but the pairing is successfully deleted. We came across this issue while testing other DApps on v4.2.1

**Comment by IsaccoSordo** [comment]
Hi @AlexandrosGounis,
I’m having a hard time trying to reproduce this issue. Here are the steps I followed:
1. Request permission, sync with the wallet, and then approve the request on the wallet side.
2. Click "Disconnect."

Is this correct?

**Comment by AlexandrosGounis** [comment]
@IsaccoSordo,
The issue is because the beacon client assumes an `IndexedDB` exists and is operable.
I noticed that my web client did not have a beacon IndexedDB instance, probably because I used the `Clear Site Data` in the past.

### To Reproduce
1. Using Chrome on desktop, go to the latest [example DApp deployment](https://8452a04057e308e40c9f01c821490fca36bef9f0--beacon-sdk.netlify.app/dapp)
2. Open the dev console → Application tab → Storage → **Clear Site Data**
3. Request permissions, approve
4. Click disconnect

The issue is that `indexedDB` may fail to be instantiated for a number of reasons (origin policies, unsupported browser, or errors during instantiation). It might be worth catching errors on instantiation, or checking if an indexedDB exists before applying transactions on it

**Comment by IsaccoSordo** [comment]
Hi @AlexandrosGounis,
as you correctly pointed out, there are situations in which `indexDB` fails to operate.
With #762, we added error checking around our "IndexedDB implementation".
Here's the lines of code which addresses your issue: [L491-#L498](https://github.com/airgap-it/beacon-sdk/blob/a4723b033a2e4a75ebd62d4798554639b488f26d/packages/beacon-dapp/src/dapp-client/DAppClient.ts#L491-#L498)

**Comment by IsaccoSordo** [comment]
The fix has been included in [v4.2.2](https://github.com/airgap-it/beacon-sdk/releases/tag/v4.2.2)
